FileIO/GmshIO/GMSHPolygonTree.cpp

```diff
@@ -86,7 +86,11 @@ void GMSHPolygonTree::insertPolyline (GeoLib::PolylineWithSegmentMarker * ply)
 			if (! ply->isSegmentMarked(k)) {
 				std::size_t seg_num(0);
 				GeoLib::Point *intersection_pnt(new GeoLib::Point);
-				while (_node_polygon->getNextIntersectionPointPolygonLine(*(ply->getPoint(k)), *(ply->getPoint(k+1)), intersection_pnt, seg_num)) {
+				while (_node_polygon->getNextIntersectionPointPolygonLine(
+				    {const_cast<GeoLib::Point*>(ply->getPoint(k)),
+				     const_cast<GeoLib::Point*>(ply->getPoint(k + 1))},
```

Comment:
In the future the internal data of the polyline should be accessed via line segment iterator instead of getPoint().